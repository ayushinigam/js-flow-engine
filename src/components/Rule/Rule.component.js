/* Rule: An accordian component which shows rule details and changes color based on status*/

import React, {PureComponent} from 'react';
import InputWithLabel from '../InputWithLabel/InputWithLabel.component';
import './Rule.styles.css'

class Rule extends PureComponent {
  state = {
    collapsed: true
  }
  STATUS_CLASSES_MAP = {
    true: 'rule-success',
    false: 'rule-failed',
    null: ''
  }
  toggleContent = () => this.setState({collapsed: !this.state.collapsed})

  render () {
    const {body, passedRuleID, failedRuleID, id, status, title} = this.props;
    return (
      <div className='rule-container'>
        <div className={`rule-header ${this.STATUS_CLASSES_MAP[status]}`} onClick={this.toggleContent}>
          <span className='rule-id'>{id}</span>
          <p className='rule-title'>{title}</p>
        </div>
        <div className={`rule-content ${this.state.collapsed ? 'rule-collapsed' : 'rule-un-collapsed'}`}>
          <InputWithLabel inputClassName='rule-body' label='Rule body' disabled value={body} />
          <div className='next-rules-container'>
            <InputWithLabel className='next-rule' label='Next rule-id if passed' disabled value={passedRuleID} />
            <InputWithLabel className='next-rule' label='Next rule-id if failed' disabled value={failedRuleID} />
          </div>
        </div>
      </div>);
  }
}
Rule.defaultProps = {
  status: null,
  value: '',
  title: 'No Title'
};

export default Rule;
