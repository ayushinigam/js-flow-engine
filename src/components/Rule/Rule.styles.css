.rule-container {
  border-radius: 5px;
  overflow: hidden;
  margin: 20px 0;
  color: #FEFFFD; }

.rule-header {
  background-color: #686968;
  color: #FEFFFD;
  cursor: pointer; }
  .rule-header .rule-id {
    background-color: #a5a5a5; }
  .rule-header .rule-id, .rule-header .rule-title {
    padding: 15px 20px;
    transition: background-color 0.5s;
    display: inline-block;
    font-size: 16px; }
  .rule-header.rule-success {
    background-color: #76A900; }
    .rule-header.rule-success .rule-id {
      background-color: #99BF3F; }
  .rule-header.rule-failed {
    background-color: #E20913; }
    .rule-header.rule-failed .rule-id {
      background-color: #EA444E; }

.rule-content {
  background-color: #929292;
  padding: 10px;
  max-height: 800px;
  transition: all 0.5s; }
  .rule-content .rule-body {
    padding-top: 20px;
    padding-bottom: 20px; }
  .rule-content .next-rules-container {
    overflow: hidden; }
  .rule-content .next-rule {
    width: 50%;
    float: left; }

.rule-collapsed {
  max-height: 0;
  padding: 0; }
