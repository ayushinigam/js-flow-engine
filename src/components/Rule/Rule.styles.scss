@import "../../variables.scss";

.rule {
  &-container {
    border-radius: 5px;
    overflow: hidden;
    margin: 20px 0;
    color: $LIGHT_FONT_COLOR;
  }
  &-header {
    background-color: $PRIMARY_HEADER_DEFAULT;
    color: $LIGHT_FONT_COLOR;
    cursor: pointer;
    .rule-id {
      background-color: $SECONDARY_HEADER_DEFAULT;
    }
    .rule-id, .rule-title  {
      padding: 15px 20px;
      transition: background-color 0.5s;
      display: inline-block;
      font-size: $FONT_LARGE;
    }
    &.rule-success {
      .rule-id {
        background-color: $SECONDARY_SUCCESS_COLOR;
      }
      background-color: $PRIMARY_SUCCESS_COLOR;
    }
    &.rule-failed {
      .rule-id {
        background-color: $SECONDARY_FAILURE_COLOR;
      }
      background-color: $PRIMARY_FAILURE_COLOR;
    }
  }
  &-content {
    background-color: $RULE_BACKGROUND;
    padding: 10px;
    max-height: 800px;
    transition: all 0.5s;
    .rule-body {
      padding-top: 20px;
      padding-bottom: 20px;
    }
    .next-rules-container {
      overflow: hidden;
    }
    .next-rule {
      width: 50%;
      float: left;
    }
  }
  &-collapsed {
    // display: none;
    max-height: 0;
    padding: 0;
  }

}
