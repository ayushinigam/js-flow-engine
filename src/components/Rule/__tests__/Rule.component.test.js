import React from 'react';
import renderer from 'react-test-renderer';
import Rule from '../Rule.component';
import {shallow} from 'enzyme';

describe('Rule component', () => {
  it('Rule: renders correctly', () => {
    const tree = renderer.create(<Rule />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('toggleContent: should toggle accordian visibility', () => {
    const wrapper = shallow(<Rule></Rule>);
    expect(wrapper.instance().state.collapsed).toBe(true);
    wrapper.instance().toggleContent();
    expect(wrapper.instance().state.collapsed).toBe(false);
  })
});
