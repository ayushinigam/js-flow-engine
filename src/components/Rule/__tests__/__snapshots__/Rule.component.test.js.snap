// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule component Rule: renders correctly 1`] = `
<div
  className="rule-container"
>
  <div
    className="rule-header "
    onClick={[Function]}
  >
    <span
      className="rule-id"
    />
    <p
      className="rule-title"
    >
      No Title
    </p>
  </div>
  <div
    className="rule-content rule-collapsed"
  >
    <div
      className="input-with-label undefined"
    >
      <p>
        Rule body
      </p>
      <input
        className="rule-body"
        disabled={true}
        value="undefined"
      />
    </div>
    <div
      className="next-rules-container"
    >
      <div
        className="input-with-label next-rule"
      >
        <p>
          Next rule-id if passed
        </p>
        <input
          className={undefined}
          disabled={true}
          value="undefined"
        />
      </div>
      <div
        className="input-with-label next-rule"
      >
        <p>
          Next rule-id if failed
        </p>
        <input
          className={undefined}
          disabled={true}
          value="undefined"
        />
      </div>
    </div>
  </div>
</div>
`;
